.github/workflows/main.yml

```diff
@@ -4,279 +4,419 @@ name: hdf5 dev CI
+        run: |
+          sh ./autogen.sh
+          mkdir "${{ runner.workspace }}/build"
+          cd "${{ runner.workspace }}/build"
+          ${{ matrix.flags }} $GITHUB_WORKSPACE/configure --enable-build-mode=${{ matrix.build_mode.autotools }} --enable-shared --enable-threadsafe --disable-hl --${{ matrix.parallel }}-parallel --${{ matrix.mirror_vfd }}-mirror-vfd --${{ matrix.direct_vfd }}-direct-vfd
+        shell: bash
+        if: (matrix.generator == 'autogen') && (matrix.thread_safe.enabled)
+
+      #
+      # CMAKE CONFIGURE
+      #
+
+      - name: CMake Configure
+        run: |
+          mkdir "${{ runner.workspace }}/build"
+          cd "${{ runner.workspace }}/build"
+          cmake ${{ matrix.generator }} -DCMAKE_BUILD_TYPE=${{ matrix.build_mode.cmake }} -DCMAKE_TOOLCHAIN_FILE=${{ matrix.toolchain }} -DBUILD_SHARED_LIBS=ON -DHDF5_ENABLE_ALL_WARNINGS=ON -DHDF5_ENABLE_PARALLEL:BOOL=${{ matrix.parallel }} -DHDF5_BUILD_CPP_LIB:BOOL=${{ matrix.cpp }} -DHDF5_BUILD_FORTRAN=${{ matrix.fortran }} -DHDF5_BUILD_JAVA=${{ matrix.java }} -DHDF5_ENABLE_MIRROR_VFD:BOOL=${{ matrix.mirror_vfd }} -DHDF5_ENABLE_DIRECT_VFD:BOOL=${{ matrix.direct_vfd }} $GITHUB_WORKSPACE
+        shell: bash
+        if: (matrix.generator != 'autogen') && (! matrix.thread_safe.enabled)
+
+
+      - name: CMake Configure (Thread-Safe)
+        run: |
+          mkdir "${{ runner.workspace }}/build"
+          cd "${{ runner.workspace }}/build"
+          cmake ${{ matrix.generator }} -DCMAKE_BUILD_TYPE=${{ matrix.build_mode.cmake }} -DCMAKE_TOOLCHAIN_FILE=${{ matrix.toolchain }} -DBUILD_SHARED_LIBS=ON -DHDF5_ENABLE_ALL_WARNINGS=ON -DHDF5_ENABLE_THREADSAFE:BOOL=ON -DHDF5_ENABLE_PARALLEL:BOOL=${{ matrix.parallel }} -DHDF5_BUILD_HL_LIB:BOOL=OFF -DHDF5_ENABLE_MIRROR_VFD:BOOL=${{ matrix.mirror_vfd }} -DHDF5_ENABLE_DIRECT_VFD:BOOL=${{ matrix.direct_vfd }} $GITHUB_WORKSPACE
+        shell: bash
+        if: (matrix.generator != 'autogen') && (matrix.thread_safe.enabled)
+
+      #
+      # BUILD
+      #
+
+      - name: Autotools Build
+        run: make
+        working-directory: ${{ runner.workspace }}/build
+        if: matrix.generator == 'autogen'
+
+      - name: CMake Build
+        run: cmake --build . --config ${{ matrix.build_mode.cmake }}
+        working-directory: ${{ runner.workspace }}/build
+        if: matrix.generator != 'autogen'
+
+      #
+      # RUN TESTS
+      #
+
+      - name: Autotools Run Tests
+        run: make check
+        working-directory: ${{ runner.workspace }}/build
+        if: (matrix.generator == 'autogen') && (matrix.run_tests)
+
+      - name: CMake Run Tests
+        run: ctest --build . -C ${{ matrix.build_mode.cmake }} -V
+        working-directory: ${{ runner.workspace }}/build
+        # Skip Debug MSVC while we investigate H5L Java test timeouts
+        if: (matrix.generator != 'autogen') && (matrix.run_tests) && ! ((matrix.name == 'Windows MSVC CMake') && (matrix.build_mode.cmake == 'Debug'))
```

Comment:
Windows Debug builds time out on H5L Java tests. We currently don't test this, so it's been hidden. We'll investigate and fix (see #2048), but in the meantime we'll just bypass.